physics/ephemeris.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <functional>
+#include <map>
+#include <memory>
+
+#include "base/not_null.hpp"
+#include "integrators/ordinary_differential_equations.hpp"
+#include "physics/continuous_trajectory.hpp"
+#include "physics/massive_body.hpp"
+#include "physics/trajectory.hpp"
+
+namespace principia {
+
+using integrators::AdaptiveStepSizeIntegrator;
+using integrators::FixedStepSizeIntegrator;
+using integrators::SpecialSecondOrderDifferentialEquation;
+
+namespace physics {
+
+template<typename Frame>
+class Ephemeris {
```

Comment:
So I think that this should ultimately just replace the existing NBodySystem (and be called NBodySystem), although we cannot do that yet as it would break everything.